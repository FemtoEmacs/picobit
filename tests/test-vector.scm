;; (define x (vector 2 4 6 8 9))
(define x '#(2 4 6 8 9))
;; (display (vector-ref x 2))
;; (vector-set! x 2 7)
;; (display (vector-ref x 2))
;; (display (bitwise-xor (vector-ref x 2) (vector-ref x 4)))
(define y (vector 1 2 3 4))
(display (+ (car x) (vector-ref x 1) (vector-ref y 3)))
