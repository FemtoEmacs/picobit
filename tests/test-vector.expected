:0080000080
:10800000FBD7230880060125811E40008112400015
:10801000810E4000810C4000810D4000800A011A51
:10802000804200028027011B8027011C8024011D43
:108030008071011F806E0120800D000280270121C8
:10804000807701228073012380240002806A00026D
:10805000807301248006012580670126800C0111B0
:1080600080400113804001148032011580270116E1
:10807000806601178069011980780002807D010FF8
:10808000806601108070010B8008010A8078010B66
:1080900002B402275202B401A85502B401A5540249
:1080A000B401965102B4019053445641500506075D
:1080B00008466457A104CEA1040540624707406208
:1080C0004263B582EED80120DAB8954321DBB68051
:1080D0000221CC9A21CE2161EE2022CFB67200EF90
:1080E00020B6800120DA9EA02607E620B558EEA033
:1080F0002607E600EF20C0B8BF20B582B6480120B1
:1081000004C8B8AEA03121B5A402B583CDDCEF021E
:10811000A034210EC5C122CD2102CDB4011F7101B1
:10812000210EC89220EF210E456221B66201042182
:10813000C2EF2002B559B75520CCB8B8A02C07E63D
:1081400020CEB51FEE20CFB6800120D296A02D07FD
:10815000E600EF20CC9DA02407E620CEB505EE205A
:10816000CFB666A107B000C6EE20B000E3EEA02DAA
:1081700007E600EF20D995A108B100C620F09C20A9
:1081800095A106B100C6A105B100C6A109B100C6FE
:10819000012007E600EF022104C79320CEEF20CF95
:1081A0002205C24172FF20EF022004C8212302CD24
:1081B000CDB401B571012204C822242302CDCDCD56
:1081C000B401C4710123B8DC21B0012D23222602A1
:1081D000CDCDCDB401D7710122B8C621B0012D217A
:1081E000242602CDCDCDB401EA710120222426023D
:1081F000CDCDCDCDB401F870002121C4232502CD11
:10820000CDB40205710122B893219220EF2298018A
:108210009420B1012D20EF21B87BB77500B86EB75F
:108220006A21B73B21B725FE21CCB8922022B68027
:108230000221CC982022CEC122CFB67420EF20EFAD
:00000001FF
